cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -353,7 +355,18 @@ public void stopAll(NamespaceId namespace)
           ProgramId program = appId.program(programRecord.getType(), programRecord.getName());
           String status = this.getStatus(program);
           if (!status.equals("STOPPED")) {
-            this.stop(program);
+            try {
+              this.stop(program);
+            } catch (IOException ioe) {
+              // ProgramClient#stop calls RestClient, which throws an IOException if the HTTP response code is 400,
```

Comment:
We could update ProgramClient#stop method to throw a `BadRequestException` or `ProgramAlreadyStopped` exception, but I didn't want to make that change in release/4.1, because that would be breaking compatibility in a patch version.

By the way, the `IOException` is not thrown from `HttpRequests` class, but from `RESTClient`:
https://github.com/caskdata/cdap/blob/develop/cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java#L132